### DISCUSSION TITLE
Add Pipes to SQL?

### DISCUSSION THREAD

**Discussion by ran-codes** [duckdb/duckdb#13599] [Ideas]
Hi just read that BigQuery added pipes - https://research.google/pubs/sql-has-problems-we-can-fix-them-pipe-syntax-in-sql/.

I think this is a 10x productivity boost in terms of writing queries. Is this possible or don't he roadmap for duckdb? (really great if it could be)

Thanks

**Comment by szarnyasg** [comment]
Hi @ran-codes, thanks for the suggestion! While possible, implementing this very likely needs a new parser infrastructure, which we are already considering, see the "Extensible Parser" point in the [DuckCon #5 status update talk](https://blobs.duckdb.org/events/duckcon5/hannes-muhleisen-mark-raasveldt-introduction-and-state-of-project.pdf#page=37). However, there is currently no ETA for when this is going to be implemented.

**Comment by BasixKOR** [comment]
SQL pipe syntax is [now available as a public preview on BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/pipe-syntax#pipe_query_syntax) today! There's also [a community extension](https://github.com/ywelsch/duckdb-psql) you could use now.

**Comment by jimdickson007** [comment]
Think link below is a different use case (making Duck part of Unix pipeline). but posting in case of general interest
https://www.linkedin.com/pulse/enhancing-duckdb-unix-pipe-integration-introducing-shellfs-conover-f0jwe/
( https://duckdb.org/community_extensions/extensions/shellfs.html )

Not a recommendation but just noticed this too : https://github.com/ywelsch/duckdb-prql

**Comment by apstndb** [comment]
- Related: https://github.com/duckdb/duckdb/discussions/14187
